### PR TITLE
[FIXED JENKINS-37814] make the icon of the script console configurable

### PR DESCRIPTION
**Pull request by rpionke**
[JENKINS-37814](https://issues.jenkins-ci.org/browse/JENKINS-37814)

discussion of the first PR (reverted) https://github.com/jenkinsci/jenkins/pull/2522

@oleg-nenashev @Vlatombe how about this approach?

### CODE REVIEW

File: core/src/main/resources/lib/hudson/scriptConsole.jelly [hunk 1]
```diff
@@ -26,12 +26,18 @@ THE SOFTWARE.
...
 -->
 <?jelly escape-by-default='true'?>
-<j:jelly xmlns:j="jelly:core" xmlns:st="jelly:stapler" xmlns:d="jelly:define" xmlns:l="/lib/layout" xmlns:t="/lib/hudson" xmlns:f="/lib/form">
+<j:jelly xmlns:j="jelly:core" xmlns:st="jelly:stapler" xmlns:d="jelly:define" xmlns:l="/lib/layout" xmlns:f="/lib/form">
   <l:layout norefresh="true" permission="${h.RUN_SCRIPTS}">
     <st:include page="sidepanel.jelly" />
 
     <l:main-panel>
-      <h1><img src="${imagesURL}/48x48/${it.icon}" width="48" height="48" alt=""/> ${%Script Console}</h1>
+      <j:if test="${it.icon == null}">
```
Comment (Vlatombe) [inline core/src/main/resources/lib/hudson/scriptConsole.jelly]: <j:choose> would be better here.

**Comment by oleg-nenashev** [comment]
The approach is a bit unreliable since the plugin providing icon may have 48x48 one.
On the other hand it may have specify full path to icon, which is not equal to imagesURL/48x/48.
So ideally it needs additional changes and integration with the icon class functionality implemented by @tfennelly .

But 👍 since it does not increase the technical debt and fixes the problem stated in JENKINS-37814.

**Comment by daniel-beck** [comment]
Is this a recent regression? If so, from when?

**Comment by oleg-nenashev** [comment]
Maybe https://github.com/rpionke/jenkins/commit/d8b1111cea8b43ae946788ce9385ac7d1264d738 (but it would mean it have never worked)

**Comment by Vlatombe** [comment]
I agree with Oleg, I think it has never worked

**Pull request merged**